rfcs/tabs-component.md

```diff
@@ -0,0 +1,150 @@
+# Tabs Component
+
+## Usage
+
+### Basic usage
+
+```js
+import * as React from 'react';
+import {Tabs, TabPanel} from 'baseui/tabs';
+
+export default () => {
+  return (
+    <div>
+      <Tabs initialState={{activeKey: '1'}}>
+        <TabPanel key="1" title="Tab 1">Tab 1 Content</Tab>
+        <TabPanel key="2" title="Tab 2">Tab 2 Content</Tab>
+        <TabPanel key="3" title="Tab 3">Tab 3 Content</Tab>
+      </Tabs>
+    </div>
+)};
+```
+
+### Advanced usage
+
+```js
+import * as React from 'react';
+import {Tabs, TabPanel, StyledTabBar, StyledTabContent, StyledTab, ORIENTATION} from 'baseui/tabs';
+import {styled} from 'baseui';
+
+CustomTabBar = styled(StyledTabBar, props => ({
+  backgroundColor: 'green'
+}));
+
+CustomTabContent = styled(StyledTabContent, props => ({
+  padding: '10px'
+}));
+
+const overrides = {
+  TabBar: CustomTabBar,
+  TabContent: CustomTabContent
+};
+
+CustomTab = styled(StyledTab, props => ({
+  borderBottom: props.$active ? `1px red` : 'none'
+}));
+
+const TabOverrides = {
+  Tab: CustomTab
+};
+
+export default () => {
+  return (
+    <div>
+      <Tabs orientation={ORIENTATION.vertical} initialState={{activeKey: '1'}} overrides={overrides}>
+        <TabPanel key="1" title="Tab 1" overrides={TabOverrides}>Tab 1 Content</Tab>
+        <TabPanel key="2" title="Tab 2" overrides={TabOverrides}>Tab 2 Content</Tab>
+        <TabPanel key="3" title="Tab 3" overrides={TabOverrides}>Tab 3 Content</Tab>
+      </Tabs>
+    </div>
+)};
+```
+
+## Exports
+
+* `Tabs`
+* `TabPanel`
+* `StyledTab`
+* `StyledTabBar`
+* `StyledTabContent`
+* `ORIENTATION`
+* `STATE_CHANGE_TYPE`
+
+## `Tabs` API
+* `children: Array<TabPanel>` - Required.
+  An array of TabPanel items.
+* `initialState: {activeKey: string}` - Optional.
+  Initial state of the component where `activeKey` is the key of the tab to be selected by default
+* `disabled: boolean` - Optional. Default is `false`
+  If set to `true` all its tabs will be disabled
+* `orientation` - Optional. Defaults to ORIENTATION.horizontal
+* `overrides: {Root, Content, TabBar}` - Optional.
```

Comment:
Above in the example we override `TabContent` should probably update example or the API description.